File: Ahorn/entities/roomChest.jl
module EeveeHelperRoomChest

using ..Ahorn, Maple

@mapdef Entity "EeveeHelper/RoomChest" RoomChest(x::Integer, y::Integer, room::String="")

const placements = Ahorn.PlacementDict(
    "Room Chest (WIP) (Eevee Helper)" => Ahorn.EntityPlacement(
        RoomChest
    )
)

function Ahorn.selection(entity::RoomChest)
    x, y = Ahorn.position(entity)

    return Ahorn.Rectangle(x - 10, y - 20, 20, 20)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::RoomChest, room::Maple.Room)
    Ahorn.drawSprite(ctx, "objects/EeveeHelper/roomChest/lid", 0, -8, jx=0.5, jy=1.0)
    Ahorn.drawSprite(ctx, "objects/EeveeHelper/roomChest/body", 0, 1, jx=0.5, jy=1.0)
end

end

File: Loenn/entities/roomChest.lua
local drawableSprite = require("structs.drawable_sprite")

local roomChest = {
    name = "EeveeHelper/RoomChest",
    depth = 5,
    placements = {
        name = "default",
        data = {
            room = "",
        }
    },
    sprite = function (room, entity)
        return {
            drawableSprite.fromTexture("objects/EeveeHelper/roomChest/lid", entity):addPosition(0, -8):setJustification(.5, 1),
            drawableSprite.fromTexture("objects/EeveeHelper/roomChest/body", entity):addPosition(0, 1):setJustification(.5, 1)
        }
    end,
}

return roomChest